tests/dragonfly/search_test.py

```diff
@@ -558,3 +562,241 @@ def make_car(producer, description, speed):
 
     for index in client.execute_command("FT._LIST"):
         client.ft(index.decode()).dropindex()
+
+
+@dfly_args({"proactor_threads": 4})
+async def test_search_race_condition_threaded_issue_5173(async_client: aioredis.Redis, df_server):
+    """
+    Alternative test using threading for true parallelism to reproduce race condition.
+    This version uses actual threads instead of asyncio tasks for maximum concurrency.
+    """
+    import threading
+    import time
+    import redis  # Sync redis client for threading
```

Comment:
it does not have to be.
you can have async functions running in threads using the following helper function:
```py
def run_in_thread(task):
        loop = asyncio.new_event_loop()
        asyncio.set_event_loop(loop)
         try:
           loop.run_until_complete(task)
       finally:
           loop.close() 
```
and then launch a thread with: `thread1 = threading.Thread(target=run_in_thread, args=(task1,), name="Thread-1")` etc

Reply:
where task1 is an `async def`